modules/images/webp-uploads/load.php

```diff
@@ -340,3 +340,130 @@ function webp_uploads_wp_get_missing_image_subsizes( $missing_sizes, $image_meta
+			continue;
+		}
+
+		if ( empty( $class_name ) ) {
+			continue;
+		}
+
+		// Make sure we use the last item on the list of matches.
+		$attachment_id = (int) end( $class_name );
+
+		if ( ! $attachment_id ) {
+			continue;
+		}
+
+		// Create an array if attachment_id has not been added to the images array.
+		if ( empty( $images[ $attachment_id ] ) ) {
+			$images[ $attachment_id ] = array();
+		}
+
+		$images[ $attachment_id ][] = $img;
+	}
+
+	foreach ( $images as $attachment_id => $images_tag ) {
+		$metadata = wp_get_attachment_metadata( $attachment_id );
+
+		foreach ( $images_tag as $img ) {
+			$content = str_replace( $img, webp_uploads_img_tag_update_mime_type( $img, $metadata ), $content );
+		}
+	}
+
+	return $content;
+}
+
+/**
+ * Finds all the urls with *.jpg and *.jpeg extension and updates with *.webp version for the provided image
+ * for the specified image sizes, the *.webp references are stored inside of each size.
+ *
+ * @since n.e.x.t
+ *
+ * @param string $image An <img> tag where the urls would be updated.
+ * @param array  $metadata An associative array with the metadata for the image.
+ * @return string The updated img tag.
+ */
+function webp_uploads_img_tag_update_mime_type( $image, array $metadata ) {
+	if ( empty( $metadata['file'] ) ) {
+		return $image;
+	}
+
+	// Creates a regular extension to find all the URLS with the provided extension for img tag.
+	preg_match_all( '/[^\s"]+\.jpe?g/i', $image, $matches );
+	if ( empty( $matches ) ) {
+		return $image;
+	}
+
+	$urls = reset( $matches );
+	// TODO: Add a filterable option to change the selected mime type. See https://github.com/WordPress/performance/issues/187.
+	$target_mime = 'image/webp';
+
+	$basename = wp_basename( $metadata['file'] );
+	foreach ( $urls as $url ) {
+		if ( isset( $metadata['file'] ) && strpos( $url, $basename ) !== false ) {
+			// TODO: we don't have a replacement for full image yet, issue. See: https://github.com/WordPress/performance/issues/174.
+			continue;
+		}
+
+		if ( empty( $metadata['sizes'] ) ) {
+			continue;
+		}
+
+		$src_filename = wp_basename( $url );
+		$extension    = wp_check_filetype( $src_filename );
+		// Extension was not set properly no action possible or extension is already in the expected mime.
+		if ( empty( $extension['type'] ) || $extension['type'] === $target_mime ) {
+			continue;
+		}
+
+		// Find the appropriate size for the provided URL.
+		foreach ( $metadata['sizes'] as $name => $size_data ) {
+			// Not the size we are looking for.
+			if ( empty( $size_data['file'] ) || $src_filename !== $size_data['file'] ) {
+				continue;
+			}
+
+			if ( empty( $size_data['sources'][ $target_mime ]['file'] ) ) {
+				continue;
+			}
+
+			// This is the same as the file we want to replace nothing to do here.
+			if ( $size_data['sources'][ $target_mime ]['file'] === $src_filename ) {
+				continue;
+			}
+
+			$image = str_replace( $src_filename, $size_data['sources'][ $target_mime ]['file'], $image );
```

Comment:
Not sure what's more performant. Now you have many individual `str_replace` calls while before you had a single `str_replace` call for all values in arrays. Was there a reason why you changed to this approach? I'd be curious to see which one is faster.

Anyway, that's kind of a micro-optimization, so not super-critical for now.

Reply:
i see the point, the previous iteration used more space (because it was creating an array to store) it had to replace every single occurrence in the array tho in the same we are doing individually inside of the loop, agree we can improve as we move forward but seems like results might be similar.